### DISCUSSION TITLE
Meaning of underscore as function parameter?

### DISCUSSION THREAD

**Discussion by ansost** [RePsychLing/SMLP2023#21] [Q&A] [answered]
In the code snippet below I tried to filter out duration values below zero with the line `filter(:wordDur => >(0), _)`. 
That line does not work without the underscore and its not a parameter mentioned in the [documentation of `filter()`](https://docs.julialang.org/en/v1/base/collections/#Base.filter) so I was wondering what it does. 

The only thing I can find about it is from here: https://c42f.github.io/Underscores.jl/stable/ but I am not sure whether that's how it is used here or how it applies here.

Full snippet:
```julia
using CSV
using DataFrames
using Chain

df = CSV.read("regression_data.csv", DataFrame)

dfFilt = @chain df begin
  filter(:wordDur => >(0), _)
  @transform!(:duration_ms = 1000 * :wordDur)
end
```

**Comment by palday** [comment, marked as answer]
The underscore is from Chain.jl and tells which argument that Chain should insert something into. By default, Chain.jl takes the first argument, which is why it isn't used for `@transform`.

You could also do `filter(:wordDur => >(0), df)` to see this applied directly to the data outside of the chain macro.
